README.md

```diff
@@ -1 +1,83 @@
-Placeholder for Checkstyle Regression Tool readme.
+# Regression-tool
+
+Regression-tool is an automation tool to do regression testing based on proposed patch(Pull Request) 
+of checkstyle project.
+
+## Setup
+
+### Clone
+
+Just clone this repository to your local.
+
+### Generate Jar
+
+Running the following command, then a `regression-tool-XX-all.jar` would be generated in the `target` folder.
+
+```bash
+$ mvn clean package -Passembly
+```
+
+## Usage
+
+```bash
+$ java -jar regression-tool-XX-all.jar -p <arg> -r <arg> [--stopAfterGenerateConfig] [-t <arg>]
+```
+
+### Arguments
+
+#### localGitRepo(r)
+
+The path of the checkstyle repository.
+
+- Required
+- One arguement
+
+#### patchBranch(p)
+
+The name of the PR branch, which would be compared with the master.
+
+- Required
+- One arguement
+
+#### stopAfterGenerateConfig
+
+Indicates that regression-tool would stop after generating config. By default, the tool would generate 
+the diff report after generating config.
+
+- Optional
+- No argument
+
+#### checkstyleTesterPath(t)
+
+The path of the checkstyle-tester directory. If you are not using `--stopAfterGenerateConfig` mode, this 
+option is required, otherwise this could be absent.
+
+- Optional
+- One argument
+
+### Example
+
+Generating regression report for branch `issue1234`.
+
+```bash
+$ java -jar target/regression-tool-XX-all.jar -r path/to/checkstyle/ -p issue1234 -t path/to/contribution/checkstyle-tester/
+```
+
+You could also use long opt.
+
+```bash
+$ java -jar target/regression-tool-XX-all.jar --localGitRepo path/to/checkstyle/ --patchBranch issue1234 --checkstyleTesterPath path/to/contribution/checkstyle-tester/
+```
+
+The tool would only generate config file and stop generating report if you are using `--stopAfterGenerateConfig` mode.
+
+```bash
+$ java -jar target/regression-tool-XX-all.jar -r path/to/checkstyle/ -p issue1234 --stopAfterGenerateConfig
+```
+
+### Output
+
+The config file would be generated in `regression-tool` directory.
+
+The report would be generated somewhere in `chechstyle-tester`. 

```

Comment:
`somewhere in chechstyle-tester.` => `in TBD`.
Remove the next line too.

Eventually we should be uploading the reports somewhere, not leaving them inside tester.
See https://github.com/checkstyle/regression-tool/issues/21 .